src/api/tasks/tasksApi.ts

```diff
@@ -59,13 +63,59 @@ export const useTasks = (subSite: string, pcolId: number) => {
 export const useAddTasks = (subSite: string, pcolId: number) => {
   const queryClient = useQueryClient();
   const pcol = usePCOL(subSite, pcolId);
+  const documents = useDocuments(subSite, String(pcol.data?.Title));
   const sendEmail = useSendEmail();
   return useMutation({
     mutationFn: async (wfDetails: WorkflowDetails) => {
       const [batched, execute] = subWebContext(subSite).batched();
       const batch = batched.web.lists.getByTitle("tasks");
       let count = 0;
 
+      const document = documents.data?.find(
+        (doc) => doc.ListItemAllFields.DocGroup === "PCOL"
+      );
+      const docRelUrl = document?.ServerRelativeUrl;
+      const attachments = documents.data?.filter(
+        (doc) => doc.ListItemAllFields.DocGroup === "Attachment"
+      );
+
+      let attachmentNames = "";
+      if (attachments) {
+        for (const att of attachments) {
+          attachmentNames += att.Name + "\n";
+        }
+      }
+
+      if (docRelUrl) {
+        PizZipUtils.getBinaryContent(
+          docRelUrl,
+          async function (error: Error, content: Data) {
+            if (error) {
+              throw error;
+            }
+            const zip = new PizZip(content);
+            const doc = new Docxtemplater(zip, {
+              paragraphLoop: true,
+              linebreaks: true,
+            });
+
+            await doc.renderAsync({
+              Attachments: attachmentNames,
+            });
+            const out = doc.getZip().generate({
+              type: "blob",
+              mimeType:
+                "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+              compression: "DEFLATE",
+            });
+
+            await subWebContext(subSite)
+              .web.getFileByServerRelativePath(docRelUrl)
+              .setContent(out);
+          }
```

Comment:
Invalidate the Documents on success, so that the PCOL shows updated by the user?